File: Day1/Go/main.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	// Define the file path
	filePath := "../input.txt"

	// Read lines from the file
	lines := ReadFile(filePath)

	// Part One Solution
	var partOne = NewDial(100, 50)
	for _, line := range lines {
		direction := string(line[0])
		steps, err := strconv.Atoi(line[1:])
		if err != nil {
			fmt.Println("Error converting steps:", err)
		}
		partOne.Turn(steps, direction)
	}
	fmt.Println("Part One - Times Reached Zero:", partOne.TimesReachedZero)

	// Part Two Solution
	var partTwo = NewDial(100, 50)
	for _, line := range lines {
		direction := string(line[0])
		steps, err := strconv.Atoi(line[1:])
		if err != nil {
			fmt.Println("Error converting steps:", err)
		}
		partTwo.TurnCountAllZeros(steps, direction)
	}
	fmt.Println("Part Two - Times Reached Zero:", partTwo.TimesReachedZero)
}

File: Day1/Python/main.py
# Import input file
from dial import Dial

with open('../input.txt', 'r') as file:
    lines = file.read().split("\n")

def __main__():
    # Initialize part one dial
    part_one = Dial(100, 50)
    for line in lines:
        direction = line[0]
        steps = int(line[1:])
        part_one.turn(steps, direction)
    print("Part 1 - Times reached zero:", part_one.timesReachedZero )

    # Initialize part two dial
    part_two = Dial(100, 50)
    for line in lines:
      direction = line[0]
      steps = int(line[1:])
      part_two.turn_check_all_steps(steps, direction)
    print("Part 2 - Times reached zero:", part_two.timesReachedZero )

if __name__ == "__main__":
    __main__()